### PR TITLE
Sendidentityprofile

### PR DESCRIPTION
**Pull request by aselalee**

### CODE REVIEW

File: updater/updater.go [hunk 1]
```diff
@@ -200,6 +251,135 @@ func upgradeIfRequired(checkURL string) (upgraded bool, err error) {
...
+
+func setRandomProfileId() int {
+	prf := Profile{}
+	nRand, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		fmt.Errorf("Error: %s\n", err.Error())
+		return 1
+	}
+	prf.Id = fmt.Sprintf("%s", nRand)
+	prf.Channel = valChannelStable
```
Comment (hiroxy) [inline updater/updater.go]: Removing L.308 and using the following code would be better.
```
prf := Profile {
    Id: fmt.Sprintf("%s", nRand),
    Channel: valChannelStable,
}
```

File: updater/updater.go [hunk 2]
```diff
@@ -200,6 +251,135 @@ func upgradeIfRequired(checkURL string) (upgraded bool, err error) {
...
+	if errSave := saveProfile(&prf); errSave != nil {
```
Comment (hiroxy) [inline updater/updater.go]: It seems no specific reason to use `errSave` instead of `err`

File: updater/updater.go [hunk 3]
```diff
@@ -200,6 +251,135 @@ func upgradeIfRequired(checkURL string) (upgraded bool, err error) {
...
+		return err
+	}
+	return ioutil.WriteFile(fn, data, 0600)
+}
+
+func setRandomProfileId() int {
+	prf := Profile{}
+	nRand, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		fmt.Errorf("Error: %s\n", err.Error())
```
Comment (hiroxy) [inline updater/updater.go]: `fmt.Errorf("Error: %v\n", err)` can be used instead

File: updater/updater.go [hunk 4]
```diff
@@ -200,6 +251,135 @@ func upgradeIfRequired(checkURL string) (upgraded bool, err error) {
...
+}
+
+func setProfileChannel(channel string) int {
+	prf := Profile{}
+	err := loadProfile(&prf)
+	prf.Channel = channel
+	if err != nil {
+		// Profile load failed. Doesn't exist or corrupted.
+		// Set id as well.
+		nRand, errRandInt := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
```
Comment (hiroxy) [inline updater/updater.go]: Creating a function to generate a random ID would be better?

**Review by hiroxy — APPROVED** [review]

File: updater/updater.go (hunk 1 above)
Comment (aselalee) [reply updater/updater.go]: done

File: updater/updater.go (hunk 2 above)
Comment (aselalee) [reply updater/updater.go]: done

File: updater/updater.go (hunk 3 above)
Comment (aselalee) [reply updater/updater.go]: done

File: updater/updater.go (hunk 4 above)
Comment (aselalee) [reply updater/updater.go]: done

**Pull request merged**